src/print/mod.rs

```diff
@@ -140,6 +140,7 @@ async fn format_rows_buf<S, I, E, O>(prn: &mut Printer<'_, O>, rows: &mut S,
         // Buffer rows up to one visual line.
         // After line is reached we get Exception::DisableFlow
     }
+    *end_of_stream = true;
```

Comment:
I thought the most appropriate fix would be to add an API to the stream itself to check if it was exhausted or not

Reply:
We accept any `Stream`, and `Stream` is a thing that `async_std` provides. So we can't add APIs. We can add another trait and require it, but I don't think this is needed. We also could use `StreamExt::fuse`, but that has overhead (i.e check of a flag on each iteration).

I still think that flag in code that uses stream is more appropriate (not because of the overhead, but because of the logic: we don't need to pull from the stream if everything is already buffered). Here it's a bit complicated because it goes from another function. But the purpose of that function factoring out is not encapsulation, but rather decreasing nesting and easier `return` statement. So propagating the flag like this is good enough for my taste.

Reply:
Alright!